### PR TITLE
Document data model

### PR DESCRIPTION
**Pull request by paulgb**

### CODE REVIEW

**Comment by paulgb** [comment]
Direct link to preview: https://pr-36.d1waxv6v9r5tig.amplifyapp.com/docs/data-model

File: docs/docs/data-model.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+DriftDB implements four actions.
+
+### `append`
+
+The `append` action results in a message being broadcast, and added to the replayable stream for its key.
+
+### `relay`
+
+The `relay` action results in a message being broadcast, but not added to the replay stream for its key. `relay` messages are still treated as part of the ordered
```
Comment (rolyatmax) [inline docs/docs/data-model.md:71]: You use the terminology "replay stream" here while using "replayable stream" above. Maybe just use one or the other?

File: docs/docs/data-model.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+The `replace` action broadcasts a message, and *replaces* the prior contents of the replay stream for just that message. You can think of `replace` as providing
+key-value semantics, where only the last value set for a given key is retained.
+
+### `compact`
+
+Unlike the other actions, `compact` must be accompanied by a sequence number. Also unlike the other actions, the compact action
+*does not* broadcast the message. Instead, it deletes all messages from the replay stream of the given key, up to and including the given sequence number.
+
+`compact` then inserts the accompanying message at the *bacK* of the replay stream for the given key, with the sequence number provided.
```
Comment (rolyatmax) [inline docs/docs/data-model.md:84]: This sentence is a little confusing to me. Maybe "the back of the replay stream" is ambiguous?

File: docs/docs/data-model.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+The `relay` action results in a message being broadcast, but not added to the replay stream for its key. `relay` messages are still treated as part of the ordered
+broadcast and given a unique sequence number, but are discarded immediately by DriftDB immediately after they are broadcast.
+
+### `replace`
+
+The `replace` action broadcasts a message, and *replaces* the prior contents of the replay stream for just that message. You can think of `replace` as providing
+key-value semantics, where only the last value set for a given key is retained.
+
+### `compact`
```
Comment (rolyatmax) [inline docs/docs/data-model.md:79]: If "replace" could optionally take a range of messages to replace, would it do the same thing as "compact"?

**Review by rolyatmax — APPROVED** [review]
Left a few comments. No blockers.

File: docs/docs/data-model.md (hunk 3 above)
Comment (paulgb) [reply docs/docs/data-model.md:79]: With compact, the message is not broadcast, but with replace it is.

Comment (rolyatmax) [reply docs/docs/data-model.md:79]: Ah yes - forgot about that

**Comment by rolyatmax** [comment]
LGTM!

**Pull request merged**